### PR TITLE
More Efficient `GridPatch` and `RandGridPatch` V2

### LINKED ISSUE #4472: `GridPatch` Performance Issue
**Describe the bug**
`GridPatch` and `RandGridPatch` are slower than their similar component `TileOnGrid`.

**To Reproduce**
```python
import numpy as np
from monai.transforms import GridPatchd
from monai.apps.pathology.transforms import TileOnGridd
from numpy.testing import assert_array_equal, assert_equal

# input configs
patch_size = 256
num_patches = 90
h, w = 10000, 1000
img = {"image": np.random.randint(0, 255, (3, h, w))}

# create components objects
patcher = GridPatchd(
    keys=["image"],
    patch_size=(patch_size, patch_size),
    num_patches=num_patches,
    sort_fn="min",
    threshold=0.999 * 3 * 255 * patch_size * patch_size,
    pad_mode=None,
    constant_values=255,
)
tiler = TileOnGridd(
    keys=["image"],
    tile_count=num_patches,
    tile_size=patch_size,
    background_val=255,
    return_list_of_dicts=True,
)

# Make sure the outputs are identical
patch_output = patcher(img)
tile_output = tiler(img)
assert_equal(len(patch_output), len(tile_output))
for p, t in zip(patch_output, tile_output):
    assert_array_equal(p["image"], t["image"])
```
```python
%%timeit -n 10 -r 10
patch_output = patcher(img)
```
```
208 ms ± 16.4 ms per loop (mean ± std. dev. of 10 runs, 10 loops each)
```
```python
%%timeit -n 10 -r 10
tile_output = tiler(img)
```
```
104 ms ± 2.38 ms per loop (mean ± std. dev. of 10 runs, 10 loops each)
```

### PR DESCRIPTION
**Pull request by drbeh**
Fixes #4472 

### Description
This PR fixes the performance issue of `GridPatch` with the minimal changes and no API change.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [x] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [x] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [x] In-line docstrings updated.

### CODE REVIEW

File: monai/apps/detection/transforms/array.py [hunk 1]
```diff
@@ -535,7 +535,8 @@ class RotateBox90(Rotate90):
     def __init__(self, k: int = 1, spatial_axes: Tuple[int, int] = (0, 1)) -> None:
         super().__init__(k, spatial_axes)
 
-    def __call__(self, boxes: NdarrayOrTensor, spatial_size: Union[Sequence[int], int]) -> NdarrayOrTensor:  # type: ignore
+    # type: ignore
+    def __call__(self, boxes: NdarrayOrTensor, spatial_size: Union[Sequence[int], int]) -> NdarrayOrTensor:
         """
```
Comment (Nic-Ma) [inline monai/apps/detection/transforms/array.py:540]: This is auto format?
Please change it to fix mypy check:
```py
def XXX(  # ignore ...
    XXX
)
```

Thanks.

Comment (drbeh) [reply monai/apps/detection/transforms/array.py]: I think so! I revert them to be as they were!

**Review by wyli — APPROVED** [review]
Integration tests worked fine as @myron suggested.

**Pull request merged**